Olaheavy2021/registrationApp
# CBP28: Student Password Reset Functionality

# Ticket Number and description
[CBP-28: Student Password Reset Functionality](https://sunkanmiolawuwogas.atlassian.net/browse/CBP-28)

## Context
As a student, I want to view my profile page, which displays all the details defined in the Student model. This will allow me to review and update my personal information easily.

## Work Done

- Validate the password input to validate the form. 
- Implement the logic to update the student's password in the database once the new password is submitted.
- Hash and securely store the new password using appropriate password-hashing techniques.

## Screenshots
<img width="639" alt="reset_password" src="https://github.com/Olaheavy2021/registrationApp/assets/86934237/f5b47e23-9cc5-4bab-bb3f-63775bd35378">



## Testing instructions
- Sign with a user
- Navigate to the reset password page on the dashboard
- Enter the required fields and save the changes
- Verify the password by signing out and signing in with the new password.

## Acceptance Criteria

- [x] The "Reset Password" page allows students to enter and confirm a new password.
- [x] The new password is securely updated in the database after successful submission.
- [x] Students receive a notification after a successful password reset.
- [x] The password reset functionality has been thoroughly tested and validated for accuracy and security.
